src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -680,7 +680,11 @@ public static function findOrNew($id, $columns = ['*'])
             return $model;
         }
 
-        return new static;
+        $model = new static;
+        
+        $model->_id = $id
```

Comment:
This won't work. You'll need something like `$model->{$model->getKeyName()} = $id`